src/NLog.StructuredLogging.Json/LoggerExtensions.cs

```diff
@@ -119,10 +120,28 @@ private static void TransferDataToLogEventProperties(LogEventInfo log, object lo
             }
         }
 
+        private static void TransferContextDataToLogEventProperties(LogEventInfo log)
+        {
+            foreach (var contextItemName in MappedDiagnosticsLogicalContext.GetNames())
+            {
+                var value = MappedDiagnosticsLogicalContext.Get(contextItemName);
```

Comment:
Maybe get the value from inside the last `if`? Then we don't "pay the price" of fetching it unless we need to use it.